core/xml.js

```diff
@@ -332,6 +338,71 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
 };
 
 /**
+ * Decode an XML DOM and create blocks on the workspace.
+ * Append the new blocks at the bottom of the existing blocks
+ * @param {!Element} xml XML DOM.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.appendDomToWorkspace = function(xml, workspace) {
+  
+  // find bottom position and alignment of the current blocks  
+  //var blocks = workspace.getAllBlocks();
+  var blocks = workspace.getTopBlocks();
+  var offsetY = 0; // container for the y of the bottom block
+  var offsetX = 0; 
+  var blocksCount = blocks.length;
+  if (blocksCount > 0) {
+	// initialize the position of the corners using position 
+	// of the first block
+	var bHW = (("height" in blocks[0])? blocks[0].getHeightWidth() : {height: 0, width: 0});
```

Comment:
@picklesrus Is the TAB width in a unit appropriate for this case, with respect to zoom? Should the getBlocksBoundingBox() be updated to include this width? It looks like it should for zoomToFit(), but we also use it in getTopLevelWorkspaceMetrics_().